projects/plugin/src/plugin/components/debrid-files/debrid-files.component.ts

```diff
@@ -1,51 +1,27 @@
 import { Component, OnInit } from '@angular/core';
 
-import { HttpClient } from '@angular/common/http';
-
-import { DebridAccountService } from '../../services/debrid-account.service';
 import { OpenSourceService } from '../../services/open-source.service';
+import { PremiumizeFolderListForm } from '../../services/premiumize/forms/folder/premiumize-folder-list.form';
 
 @Component({
   selector: 'wk-debrid-files',
   templateUrl: './debrid-files.component.html',
   styleUrls: ['./debrid-files.component.scss']
 })
 export class DebridFilesComponent implements OnInit {
-  private key = '';
-
-  private base_link = 'https://www.premiumize.me/api';
-
-  private api_folderAll = '/folder/list';
+  public response;
 
-  public folder;
-
-  constructor(private http: HttpClient, private debridAccountService: DebridAccountService, private openSourceService: OpenSourceService) {}
+  constructor(private openSourceService: OpenSourceService) {}
 
   ngOnInit() {
-    this.debridAccountService
-      .getPremiumizeSettings()
-      .then((settings) => {
-        this.key = settings.apiKey;
-      })
-      .then(() => {
-        this.listAll('');
-      });
+    this.listAll('');
   }
 
   public async listAll(folderID) {
-    let data;
-
-    if (folderID !== '') {
-      data = await this.http.get(this.base_link + this.api_folderAll + '?id=' + folderID + '&apikey=' + this.key).toPromise();
-    } else {
-      data = await this.http.get(this.base_link + this.api_folderAll + '?apikey=' + this.key).toPromise();
-    }
-
-    this.folder = data;
+    this.response = await PremiumizeFolderListForm.submit(folderID).toPromise();
   }
 
-  //TODO
   public openLink(link) {
-    this.openSourceService.open(link, 'open-kodi', null);
+    //this.openSourceService.openKodi([link]);
```

Comment:
I tried to use the 'openKodi' function from the 'OpenSourceService'. 
But it's a private function so I can only call it if I change it to public, but i don't think thats the best solution. 

How do I use this function?

Reply:
Actually you shouldn't try to open the link directly in kodi because users may want to open it via another method like vlc, chromecast... so you should create a `StreamLinkSource` and open it via `openSourceService.openStreamLinkSource` and will do the rest. 

I don't know what the folder api returns, but I guess you got something similar to here: https://github.com/wako-unofficial-addons/helios/blob/master/projects/plugin/src/plugin/queries/debrids/premiumize/premiumize-get-stream-link.query.ts#L12 so you already have what you need to fill `StreamLinkSource.streamLinks` array.

Btw, FYI, I don't know if you already have tried to use your plugin on your phone, but you can do it.  Run the command `npm run serve:wako-like` with your own computer's IP (https://github.com/wako-unofficial-addons/helios/blob/master/package.json#L8) and then you can install the plugin inside wako with this URL: `http://X.X.X.X:4200/assets/plugins/manifest.json where X.X.X.X` is your own IP address. More information on the wako starter kit https://github.com/wako-app/addon-starter-kit#test-your-add-on-on-wako

Reply:
Yes the Api returns similar values like the example you linked.

I dont understand the difference between `StreamLinkSource` and `StreamLink`. 

I tried to create a new `StreamLinkSource` and a `StreamLink` Object and filled all the values. 
Then pushed the `StreamLink` into the `StreamLinkSource.streamLink` array.

`openSourceService.openStreamLinkSource` also need a `SourceQuery`, which I don't know how to use.

Reply:
A `StreamLinkSource`  is the container of a `StreamLink` array. For instance a torrent will be represented by a `StreamLinkSource`, while all the files it contains will be the `StreamLink` array. At the end, helios will open a `StreamLink`  

For the `SouceQuery` here, fill only the mandatory property `category`. If you don't know what to set, set it to `movie`